Common/SymbolRepresentation.cs

```diff
@@ -338,40 +339,39 @@ public static Symbol ParseOptionTickerOSI(string ticker, SecurityType securityTy
         /// <returns>Symbol object for the specified OSI option ticker string</returns>
         public static Symbol ParseOptionTickerOSI(string ticker, SecurityType securityType, OptionStyle optionStyle, string market)
         {
-            var optionTicker = ticker.Substring(0, 6).Trim();
-            var expiration = DateTime.ParseExact(ticker.Substring(6, 6), DateFormat.SixCharacter, null);
-            OptionRight right;
-            if (ticker[12] == 'C' || ticker[12] == 'c')
-            {
-                right = OptionRight.Call;
-            }
-            else if (ticker[12] == 'P' || ticker[12] == 'p')
-            {
-                right = OptionRight.Put;
-            }
-            else
+            
+            var match = Regex.Match(ticker, @"^([A-Z]+)\s*(\d{6})([CP])(\d{8})$", RegexOptions.IgnoreCase);
+            if (!match.Success)
```

Comment:
Let's define this as a private readonly static field at the top, compiled regex, so we reuse it 👍

Reply:
sure will work onit